mobile/src/calendar/event-dialog/dayoff-requested-dialog.tsx

```diff
@@ -0,0 +1,39 @@
+/******************************************************************************
+ * Copyright (c) Arcadia, Inc. All rights reserved.
+ ******************************************************************************/
+
+import { Component } from 'react';
+import { EventDialogBase } from './event-dialog-base';
+import React from 'react';
+import { connect } from 'react-redux';
+import { Action, Dispatch } from 'redux';
+import { closeEventDialog } from '../../reducers/calendar/event-dialog/event-dialog.action';
+
+//============================================================================
+interface DayoffRequestedDialogDispatchProps {
+    closeDialog: () => void;
+}
+
+//============================================================================
+class DayoffRequestedDialogImpl extends Component<DayoffRequestedDialogDispatchProps> {
+    public render() {
+        return <EventDialogBase
+            title={'Dayoff requested'}
+            text={'Your dayoff request will be shortly reviewed by your supervisor'}
+            icon={'dayoff'}
+            acceptLabel={'Ok'}
+            onAcceptPress={this.props.closeDialog}
+            onClosePress={this.props.closeDialog} />;
+    }
+}
+
+//----------------------------------------------------------------------------
+const mapDispatchToProps = (dispatch: Dispatch<Action>): DayoffRequestedDialogDispatchProps => ({
+    closeDialog: () => {
+        dispatch(closeEventDialog());
+    },
+});
+
+const DayoffRequestedDialog = connect(undefined, mapDispatchToProps)(DayoffRequestedDialogImpl);
+
+export default DayoffRequestedDialog;
```

Comment:
please avoid default exports (khm, how I forgot to enable it https://palantir.github.io/tslint/rules/no-default-export/).
Three reasons:
1) non-default export allows you to change a variable name in a non-obvious way (otherwise it requires 'as') - and if initially IDE imports in it a proper way, later name changes might not perform depending on personal IDE settings
2) Tree-shaking works way better (the thing that loads only necessary modules) - it's not actually the problem in react-native, I just give you a generic web example
3) In JS/TS world, statements like that are often used by old/deprecated module systems that behave differently, because they actually imported a module, so it's confusing for a reader.

Reply:
I mean, there is no serious technical reason to use non-default over default, I just explain why this is preferable way :)